backend/inspirehep/disambiguation/tasks.py

```diff
@@ -142,36 +134,34 @@ def match_literature_author(author, record):
         matched_records = match_literature_author_with_config(
             author_matcher_data, config
         )
-        matched_refs_ids = get_reference_and_ids_for_literature_author_match(
+        matched_refs = get_author_references_for_literature_author_match(
             matched_records
         )
-        matched_author_data = get_reference_and_bai_if_unambiguous_match(
-            matched_refs_ids
-        )
+        matched_author_reference = get_reference_if_unambiguous_match(matched_refs)
 
-        if matched_author_data:
+        if matched_author_reference:
             break
```

Comment:
wouldn't be sufficient to check the size? instead of using `get_reference_if_unambiguous_match`